### PR TITLE
Small updates for core tests to work with the main automation.

### PR DESCRIPTION
**Pull request by gearama**
Small bug fixes that caused failures when running in the pipeline.
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ namespace Azure { namespace Perf { namespace Test {
     {
       // Increment the counter and fetch the value, then remove 1 to get the previous-increment
       // value
-      auto instanceCount = _detail::DelayTestInstanceCount.fetch_add(1) - 1;
```
Comment (LarryOsterman) [inline sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp:59]: I'm a bit confused here. The comment says "increment the counter", but the documentation for `std::atomic::fetch_add` says that it adds the first parameter to the value, in which case the new code is adding 0 to the `DelayInstanceCount`.

File: sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp [hunk 2]
```diff
@@ -67,7 +67,9 @@ namespace Azure { namespace Perf { namespace Test {
      * @brief The test definition
      *
      */
-    void Run(Azure::Core::Context const&) override { std::this_thread::sleep_for(m_delay); }
```
Comment (LarryOsterman) [inline sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp:70]: What changed to cause this? I would think that clang-format would be upset and one of these two forms.

File: sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp (hunk 1 above)
Comment (gearama) [reply sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp:59]: whatever the documentation says it is wrong, under the debugger the fetc_add 1 return 0

Comment (gearama) [reply sdk/core/perf/test/inc/azure/perf/test/delay_test.hpp:59]: and really the point of the test is to do a sleep , not to waste time with ftch add

**Review by vhvb1989 — APPROVED** [review]
delay test is not super important.
If it is giving you issues, you can also remove it entirely

**Pull request merged**